generator/src/ServiceGenerator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Service\Generator;
+
+class ServiceGenerator
+{
+    const PREFERRED = [
+    'admin:datatransfer_v1' => true,
+    'admin:directory_v1' => true,
+    'discovery:v1' => false,
+    ];
+
+    public $path = __DIR__ . "/../../../../../src/Google/Service";
```

Comment:
I believe the latest revision broke the `generate-all` script, this path should be updated to:

```php
public $path = __DIR__ . "/../../../src/Google/Service";
```

It may be worthwhile to add a small test to cover these scenarios :).

Reply:
Another item of note, I receive the following notice when running `generate-all`:

`Undefined index: itemId in /Users/dsupplee/dev/google-api-php-client-services/generator/src/Method.php on line 71`

Reply:
I haven't been able to reproduce this Undefined index notice yet.

Reply:
Yes.

Reply:
Thanks Matt, looks like chat is the culprit.

```
chat:v1 assert(): assert(isset($method['parameters'][$movedKey])) failed in /Users/dsupplee/dev/google-api-php-client-services/generator/src/Method.php on line 70
chat:v1 Undefined index: itemId in /Users/dsupplee/dev/google-api-php-client-services/generator/src/Method.php on line 71
```